### ISSUE TITLE
Bad preprocessor expansion of sketch results in illegal c++ code. [imported]

### ISSUE THREAD

**Issue by cmaglie** [arduino/Arduino#297]
This is [Issue 297](http://code.google.com/p/arduino/issues/detail?id=297) moved from a Google Code project.
Added by 2010-07-22T00:52:26.000Z by [GorillaCoder](http://code.google.com/u/GorillaCoder/).
Please review that bug for more context and additional comments, but update this bug.

Original labels: Type-Defect, Priority-Medium

### Original description

Try to compile the below sketch:

```c++
    class SleepCycle  {
    public:

        SleepCycle( const char* name );
        int foo;
    }; // class SleepCycle

    SleepCycle::SleepCycle( const char* name )
    :   foo    ( 0 )
    {
    }
```

The c++ compilation fails because the generated c++ is bad:

```c++
    include "WProgram.h"
    foo    ( 0 ); 
    class SleepCycle  {
    public:
    
        SleepCycle( const char* name );
        int foo;
    }; // class SleepCycle
    
    SleepCycle::SleepCycle( const char* name )
    :   foo    ( 0 )
    {   
    }   
```

It can be fixed by removing the spaces between the name and paren in the initializer for foo.
    :   foo( 0 )

**Comment by ffissore** [comment]
Posted answer on googlecode

**Comment by ffissore** [comment]
Fixed by #3779
